### ISSUE TITLE
ta_test fails in eigen_suite if CMAKE_BUILD_TYPE=Release and > 1 MPI process

### ISSUE THREAD

**Issue by mbanck** [ValeevGroup/tiledarray#106]
I'm getting exceptions when I build TA with ```CMAKE_BUILD_TYPE=Release``` (to which I've switched due to the MIPS linker errors) and run the test suite with more than one MPI process (I'm using MPICH):

```
tiledarray-0.6.0/obj-x86_64-linux-gnu/tests$ mpiexec -np 2 ./ta_test  -t eigen_suite
MADNESS runtime initialized with 3 threads in the pool and affinity -1 -1 -1
Running 8 test cases...
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(154): error: in "eigen_suite/matrix_to_array": exception TiledArray::Exception expected but not raised
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(185): error: in "eigen_suite/vector_to_array": exception TiledArray::Exception expected but not raised
Running 8 test cases...
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(154): error: in "eigen_suite/matrix_to_array": exception TiledArray::Exception expected but not raised
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(185): error: in "eigen_suite/vector_to_array": exception TiledArray::Exception expected but not raised
A madness exception occurred. Place a break point at madness::exception_break to debug.
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
A madness exception occurred. Place a break point at madness::exception_break to debug.
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
A madness exception occurred. Place a break point at madness::exception_break to debug.
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
A madness exception occurred. Place a break point at madness::exception_break to debug.
unknown location(0): fatal error: in "eigen_suite/array_to_matrix": tbb::captured_exception: MADNESS ASSERTION FAILED: "/usr/include/madness/world/future.h"(212)
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(240): last checkpoint
unknown location(0): fatal error: in "eigen_suite/array_to_matrix": tbb::captured_exception: MADNESS ASSERTION FAILED: "/usr/include/madness/world/future.h"(212)
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(240): last checkpoint
unknown location(0): fatal error: in "eigen_suite/array_to_vector": tbb::captured_exception: MADNESS ASSERTION FAILED: "/usr/include/madness/world/future.h"(212)
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(311): last checkpoint
!!MADNESS ERROR: Exception thrown in WorldTaskQueue::fence() with 53 pending task(s)
unknown location(0): fatal error: in "eigen_suite/array_to_vector": tbb::captured_exception: MADNESS ASSERTION FAILED: "/usr/include/madness/world/future.h"(212)
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(311): last checkpoint
A madness exception occurred. Place a break point at madness::exception_break to debug.
A madness exception occurred. Place a break point at madness::exception_break to debug.
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
A madness exception occurred. Place a break point at madness::exception_break to debug.
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
terminate called after throwing an instance of 'tbb::captured_exception'
  what():  MADNESS ASSERTION FAILED: "/usr/include/madness/world/future.h"(212)

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 28755 RUNNING AT barriere
=   EXIT CODE: 6
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Aborted (signal 6)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
```

Is this reproducable? To be clear, this exception isn't thrown if ```CMAKE_BUILD_TYPE``` is ```none``` (whatever that means).

**Comment by evaleev** [comment]
the line number in this error message

```
/home/mbanck/src/tiledarray-0.6.0/tests/eigen.cpp(154): error: in "eigen_suite/matrix_to_array": exception TiledArray::Exception expected but not raised
```

suggests that the source is likely out of date. this was fixed in a9e3e3cc4d765910ab0f2dbdc8d9247ee729d766

**Comment by mbanck** [comment]
Oh yeah, this was with the 0.6.0 release. Might make sense to roll 0.6.1 then?

**Comment by evaleev** [comment]
Will release after #107 is resolved, @xwang862 what's the status of #107?

**Comment by xwang862** [comment]
#107 should be resolved by 1fceb55.
